stats/views.py

```diff
@@ -21,99 +19,123 @@
-        users.append({'date': (time.mktime(ref_date.timetuple())), 'n': real_tigausers.filter(registration_time__lte=ref_date).count()})
-        ref_date += timedelta(hours=168)
-    # now set final day as current time
-    site_reports.append({'date': time.mktime(end_date.timetuple()), 'n': real_reports.filter(type='site', creation_time__lte=ref_date).count()})
-    adult_reports.append({'date': time.mktime(end_date.timetuple()), 'n': real_reports.filter(type='adult', creation_time__lte=ref_date).count()})
-    users.append({'date': time.mktime(end_date.timetuple()), 'n': real_tigausers.filter(registration_time__lte=ref_date).count()})
-    context = {'users': users, 'site_reports': site_reports, 'adult_reports': adult_reports}
-    return render(request, 'stats/chart.html', context)
+    def get_cumsum(qs, datetime_fieldname: str, trunc_by: str) -> dict:
+        qs = qs.annotate(
+            week_epoch=Extract(
+                Trunc(datetime_fieldname, kind=trunc_by),
+                'epoch'
+            )
+        ).values('week_epoch').annotate(
+            n=Count(1)
+        ).order_by('week_epoch')
+
+        cumsum = 0
+        result = []
+        for obj in qs.iterator():
+            cumsum += obj['n']
+            result.append({
+                'date': obj['week_epoch'],
+                'n': cumsum
+            })
+        return result
 
-def get_most_recently_validated_report(slug):
-    completed_annot = ExpertReportAnnotation.objects.filter(validation_complete=True).filter(user__username=slug).extra(order_by=['-last_modified'])
-    if len(completed_annot) > 0:
-        latest_uncomplete_annotation = completed_annot.first()
-        return latest_uncomplete_annotation.created
-    return None
+    context = {
+        'users': get_cumsum(
+            qs= TigaUser.objects,
+            datetime_fieldname='registration_time',
+            trunc_by='week'
+        ),
+        'site_reports': get_cumsum(
+            qs=Report.objects.filter(type=Report.TYPE_SITE),
+            datetime_fieldname='server_upload_time',
+            trunc_by='week'
+        ),
+        'adult_reports': get_cumsum(
+            qs=Report.objects.filter(type=Report.TYPE_ADULT),
+            datetime_fieldname='server_upload_time',
+            trunc_by='week'
+        )
+    }
+    return render(request, 'stats/chart.html', context)
 
 @api_view(['GET'])
 def workload_pending_per_user(request):
     if request.method == 'GET':
         user_slug = request.query_params.get('user_slug', -1)
-        queryset = User.objects.all()
-        user = get_object_or_404(queryset, username=user_slug)
-        current_pending = ExpertReportAnnotation.objects.filter(user=user).filter(validation_complete=False).filter(report__type='adult')
+        user = get_object_or_404(User, username=user_slug)
+
+        if not hasattr(user, 'userstat'):
```

Comment:
Changed this check because  the former check threw an Exception if userstat was not present